### PR TITLE
Issue 65 - Remove foldercontent override

### LINKED ISSUE #65: make folder contents view override optional
It's rather annoying that the uploader is always present at the top of folder contents view. Would be better if it was triggered by a marker interface.

### PR DESCRIPTION
**Pull request by rodfersou**
closes #65

### CODE REVIEW

**Comment by rodfersou** [comment]
@hvelarde green

**Comment by rodfersou** [comment]
@petri @idgserpro could you please help reviewing this pull request?

File: src/collective/upload/browser.py [hunk 1]
```diff
@@ -211,13 +212,15 @@ def render(self):
...
     def registry_config(self):
         config = {}
-        registry = getUtility(IRegistry)
-        settings = registry.forInterface(IUploadSettings)
-        config['extensions'] = str(settings.upload_extensions.replace(' ', '').replace(',', '|'))
-        config['max_file_size'] = settings.max_file_size
-        config['resize_max_width'] = settings.resize_max_width
-        config['resize_max_height'] = settings.resize_max_height
-
+        record = dict(name='upload_extensions', interface=IUploadSettings)
```
Comment (idgserpro) [inline src/collective/upload/browser.py]: ```interface``` is available only in new versions of plone.api. It's not necessary pin in setup.py?

File: src/collective/upload/tests/test_viewlets.py [hunk 2]
```diff
@@ -13,21 +13,19 @@
 import unittest
 
 
-NAME = 'collective.upload.tmpls'
```
Comment (idgserpro) [inline src/collective/upload/tests/test_viewlets.py]: Why not use the constant?

File: src/collective/upload/viewlets.py [hunk 3]
```diff
@@ -16,3 +17,10 @@ def enabled(self):
         if context_state.is_default_page():
             context = context.aq_parent
         return IFolderish.providedBy(context)
+
+
+class ContentsViewlet(ViewletBase):
```
Comment (idgserpro) [inline src/collective/upload/viewlets.py]: Perhaps a more intuitive name. Something like MultipleFileUploadViewlet

File: src/collective/upload/viewlets.zcml [hunk 4]
```diff
@@ -15,4 +15,13 @@
         layer=".interfaces.IUploadBrowserLayer"
         />
 
+    <browser:viewlet
+        name="collective.upload.contents"
```
Comment (idgserpro) [inline src/collective/upload/viewlets.zcml]: Perhaps a more intuitive name.

File: src/collective/upload/tests/test_viewlets.py (hunk 2 above)
Comment (rodfersou) [inline src/collective/upload/tests/test_viewlets.py]: Before this test file was just for one viewlet, now it is used for two viewlets..

I can add a local variable to use instead.

File: src/collective/upload/viewlets.zcml (hunk 4 above)
Comment (rodfersou) [inline src/collective/upload/viewlets.zcml]: how about ``collective.upload.widget`` ?

File: src/collective/upload/browser.py (hunk 1 above)
Comment (rodfersou) [inline src/collective/upload/browser.py]: Nice catch, I'll pin it

Comment (rodfersou) [inline src/collective/upload/browser.py]: Done

Comment (hvelarde) [inline src/collective/upload/browser.py]: this is way more complex and harder to read that it was before; I'm -1 on this change.

File: src/collective/upload/tests/test_controlpanel.py [hunk 5]
```diff
@@ -60,28 +59,42 @@ class RegistryTestCase(unittest.TestCase):
 
     def setUp(self):
         self.portal = self.layer['portal']
-        self.registry = getUtility(IRegistry)
-        self.settings = self.registry.forInterface(IUploadSettings)
         setRoles(self.portal, TEST_USER_ID, ['Manager'])
 
+    def _get_setting(self, setting):
```
Comment (hvelarde) [inline src/collective/upload/tests/test_controlpanel.py]: -1 also on all of these changes in these module for the same reasons expressed above.

File: src/collective/upload/tests/test_viewlets.py [hunk 6]
```diff
@@ -39,7 +38,8 @@ def test_manager_viewlet_is_present_in_folder(self):
         self.assertTrue(manager is not None)
 
         manager.update()
-        viewlet = [v for v in manager.viewlets if v.__name__ == NAME]
+        viewlet = [
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: is this line above 80 chars?

File: src/collective/upload/tests/test_viewlets.py [hunk 7]
```diff
@@ -50,7 +50,8 @@ def test_manager_viewlet_is_present_in_portal(self):
...
+            v for v in manager.viewlets if v.__name__ == self.name]
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: is this line above 80 chars?

File: src/collective/upload/tests/test_viewlets.py [hunk 8]
```diff
@@ -61,7 +62,8 @@ def test_anonymous_viewlet_is_not_present_in_folder(self):
         self.assertTrue(manager is not None)
 
         manager.update()
-        viewlet = [v for v in manager.viewlets if v.__name__ == NAME]
+        viewlet = [
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: is this line above 80 chars?

File: src/collective/upload/tests/test_viewlets.py [hunk 9]
```diff
@@ -72,5 +74,50 @@ def test_anonymous_viewlet_is_not_present_in_portal(self):
...
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: is this line above 80 chars?

File: src/collective/upload/tests/test_viewlets.py [hunk 10]
```diff
@@ -72,5 +74,50 @@ def test_anonymous_viewlet_is_not_present_in_portal(self):
...
+            request=self.request
+        )
+        self.assertNotIn('<form id="fileupload"', view())
+
+    def test_anonymous_viewlet_is_not_present(self):
+        logout()
+        view = View(self.folder, self.request)
+        manager = queryMultiAdapter((self.folder, self.request, view),
+                                    IViewletManager, 'plone.abovecontentbody')
+        self.assertTrue(manager is not None)
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: use `self.assertIsNotNone()`

File: src/collective/upload/tests/test_viewlets.py [hunk 11]
```diff
@@ -72,5 +74,50 @@ def test_anonymous_viewlet_is_not_present_in_portal(self):
...
+
+        manager.update()
+        viewlet = [
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: is this line above 80 chars?

File: src/collective/upload/viewlets.py [hunk 12]
```diff
@@ -16,3 +17,10 @@ def enabled(self):
         if context_state.is_default_page():
             context = context.aq_parent
         return IFolderish.providedBy(context)
+
+
+class MultipleFileUploadViewlet(ViewletBase):
+
+    def enabled(self):
```
Comment (hvelarde) [inline src/collective/upload/viewlets.py]: let's make this configurable in the control panel configlet.

File: src/collective/upload/tests/test_viewlets.py [hunk 13]
```diff
@@ -39,7 +38,8 @@ def test_manager_viewlet_is_present_in_folder(self):
         self.assertTrue(manager is not None)
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: use `self.assertIsNotNone()`

File: src/collective/upload/tests/test_viewlets.py [hunk 14]
```diff
@@ -50,7 +50,8 @@ def test_manager_viewlet_is_present_in_portal(self):
...
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: use `self.assertIsNotNone()`

File: src/collective/upload/tests/test_viewlets.py [hunk 15]
```diff
@@ -61,7 +62,8 @@ def test_anonymous_viewlet_is_not_present_in_folder(self):
...
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: use `self.assertIsNotNone()`

File: src/collective/upload/tests/test_viewlets.py [hunk 16]
```diff
@@ -72,5 +74,50 @@ def test_anonymous_viewlet_is_not_present_in_portal(self):
...
```
Comment (hvelarde) [inline src/collective/upload/tests/test_viewlets.py]: use `self.assertIsNotNone()`

File: CHANGES.rst [hunk 17]
```diff
@@ -1,7 +1,8 @@
 Changelog
 ---------
 
-There's a frood who really knows where his towel is.
```
Comment (hvelarde) [inline CHANGES.rst]: why did you removed my joke?

Comment (rodfersou) [inline CHANGES.rst]: ops

File: src/collective/upload/viewlets.zcml (hunk 4 above)
Comment (idgserpro) [inline src/collective/upload/viewlets.zcml]: Something like collective.upload.multiplemfileupload

File: src/collective/upload/tests/test_viewlets.py [hunk 18]
```diff
@@ -13,21 +13,20 @@
...
 
 
-NAME = 'collective.upload.tmpls'
-
-
-class ViewletTest(unittest.TestCase):
+class TmplsViewletTest(unittest.TestCase):
 
     layer = INTEGRATION_TESTING
+    name = 'collective.upload.tmpls'
```
Comment (idgserpro) [inline src/collective/upload/tests/test_viewlets.py]: Change name here too.

Comment (idgserpro) [inline src/collective/upload/tests/test_viewlets.py]: ops. is correct

File: src/collective/upload/viewlets.py (hunk 12 above)
Comment (rodfersou) [inline src/collective/upload/viewlets.py]: in this PR?

File: src/collective/upload/tests/test_viewlets.py (hunk 10 above)
Comment (rodfersou) [inline src/collective/upload/tests/test_viewlets.py]: Now I understand.. to use ``getMultiAdapter`` instead of ``queryMultiAdapter``

**Comment by hvelarde** [comment]
@petri could you please check if this solves your use case? I'm about to merge and make a new release.

File: src/collective/upload/viewlets/contents.pt [hunk 19]
```diff
@@ -0,0 +1,5 @@
+<tal:define define="checkPermission nocall: context/portal_membership/checkPermission"
```
Comment (hvelarde) [inline src/collective/upload/viewlets/contents.pt]: @rodfersou you're not checking permissions anywhere.

**Comment by rodfersou** [comment]
@hvelarde I forgot pin ``plone.api`` version after my first code refactor

**Pull request merged**